MapboxNavigation/VisualInstructionComponent.swift

```diff
@@ -0,0 +1,40 @@
+import UIKit
+import SDWebImage
+import MapboxDirections
+
+extension VisualInstructionComponent {
+    
+    static let scale = UIScreen.main.scale
+    
+    func shieldKey() -> String {
+        return "\(imageURL!.absoluteString)-\(VisualInstructionComponent.scale)"
+    }
+    
+    func cachedShield(_ shieldKey: String) -> UIImage? {
+        return SDImageCache.shared().imageFromCache(forKey: shieldKey)
+    }
+    
+    func shieldImage(height: CGFloat, completion: @escaping (UIImage?) -> Void) {
+        guard let imageURL = imageURL else { return }
+        let shieldKey = self.shieldKey()
+        
+        if let cachedImage = self.cachedShield(shieldKey) {
+            completion(cachedImage)
+            return
+        }
+        
+            
+        SDWebImageDownloader.shared().downloadImage(with: imageURL, options: [], progress: nil, completed: { (image, data, error, successful) in
+            guard let imageData = data else { return }
+            
+            guard let downscaledImage = UIImage(data: imageData, scale: VisualInstructionComponent.scale) else {
+                completion(nil)
+                return
+            }
+            
+            SDImageCache.shared().store(downscaledImage, forKey: shieldKey, toDisk: true, completion: nil)
+            completion(downscaledImage)
```

Comment:
This completion should be moved to inside the `store(…completion:)` to guarantee that the image is cached before we reconstruct the instruction here:

https://github.com/mapbox/mapbox-navigation-ios/blob/8e7f467edcd6990795c23edb08c9e6af31c46762/MapboxNavigation/InstructionLabel.swift#L41-L44